internal/gh/projects.go

```diff
@@ -0,0 +1,23 @@
+package gh
+
+// ProjectsV1Support provides type safety and readability around whether or not Projects v1 is supported
+// by the targeted host.
+//
+// It is a sealed type to ensure that consumers must use the exported ProjectsV1Supported and ProjectsV1Unsupported
+// variables to get an instance of the type.
+type ProjectsV1Support interface {
+	sealed()
```

Comment:
As I self-review I realise the term `sealed` may not be familiar to everyone. What we're doing here is having an unexported method on the interface so that only types in this module can implement the interface.

In practice, we probably don't need it because it's unlikely anyone is going to come along and try to add another concrete implementation but it's a signal to any reader that the only options that make sense are right here.

Reply:
Would love to continue the discussion offline with examples of this use in other projects and why it is absolutely needed in those situations versus an option available.

Reply:
I don't know what you mean by "an option available". Let's definitely continue the discussion though.